plugins/airtable/src/data.ts

```diff
@@ -272,102 +291,151 @@ export async function inferFields(collection: ManagedCollection, table: Airtable
+            return {
+                value: value[0].url,
+                type: "string",
+            }
 
         case "collectionReference":
             if (!Array.isArray(value)) return null
             if (value.length === 0) return null
-            return value[0]
+            return {
+                value: value[0],
+                type: "collectionReference",
+            }
 
         case "multiCollectionReference":
             if (!Array.isArray(value)) return null
-            return value
+            return {
+                value,
+                type: "multiCollectionReference",
+            }
 
         case "date":
             if (typeof value === "string") {
                 const date = new Date(value)
-                return isNaN(date.getTime()) ? null : date.toISOString()
+                if (isNaN(date.getTime())) return null
+                return {
+                    value: date.toISOString(),
+                    type: "date",
+                }
             }
             return null
 
         case "string":
+            if (typeof value !== "string") return null
+            return {
+                value,
+                type: "string",
+            }
+
         case "color":
-            return typeof value === "string" ? value : null
+            if (typeof value !== "string") return null
+            return {
+                value,
+                type: "color",
+            }
 
         case "formattedText":
-            return typeof value === "string" ? richTextToHTML(value) : null
+            if (typeof value !== "string") return null
+            return {
+                value: richTextToHTML(value),
+                type: "formattedText",
+            }
 
         case "enum": {
             if (typeof value !== "string") return null
             const choice = fieldSchema.cases.find(choice => choice.name === value)
-            return choice?.id ?? null
+            if (!choice) return null
+            return {
+                value: choice.id,
+                type: "enum",
+            }
         }
 
         case "number":
-            return typeof value === "number" ? value : null
+            if (typeof value !== "number" || Number.isNaN(value)) return null
+            return {
+                value,
+                type: "number",
+            }
 
         default:
-            return value
+            return null
     }
 }
 
 export async function getItems(dataSource: DataSource, slugFieldId: string) {
     const items = await fetchRecords(dataSource.baseId, dataSource.tableId)
     const fieldsById = new Map(dataSource.fields.map(field => [field.id, field]))
-    const itemsData: ({ id: string } & Record<string, unknown>)[] = []
+    const itemsData: { id: string; slugValue: string; fieldData: FieldDataInput }[] = []
 
     for (const item of items) {
-        const itemData: Record<string, unknown> = {}
+        const itemData: FieldDataInput = {}
```

Comment:
Nit: If `fieldData` below becomes `fieldDataEntry`, then this could be just `fieldData` maybe?